scanner/src/main/kotlin/Scanner.kt

```diff
@@ -234,7 +234,13 @@ class Scanner(
      * Run package scanners for packages with incomplete scan results.
      */
     private fun runPackageScanners(controller: ScanController, context: ScanContext) {
-        controller.getPackagesConsolidatedByProvenance().forEach { (provenance, packages) ->
+        val packagesByProvenance = controller.getPackagesConsolidatedByProvenance()
+
+        packagesByProvenance.onEachIndexed { index, entry ->
```

Comment:
FYI: There is no `forEachIndexed` for maps.

Reply:
You could still do `onEachIndexed { index, (provenance, packages) ->`.

Reply:
Ah, great! I was pretty sure I checked this some time ago, and I believe back then that syntax was not supported. But good to see that it is now!